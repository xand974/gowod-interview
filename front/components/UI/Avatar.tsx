import React from "react";
import { StyleSheet, Text, View } from "react-native";
import styled from "styled-components/native";
import { WHITE } from "assets/styles/@core.style";
import { DEFAULT_IMG } from "mock/data";

const TopInfosContainer = styled.View`
  flex-direction: row;
  align-items: center;
  margin: 10px 0;
`;

const RoundedImageContainer = styled.View`
  flex: 1;
  width: 100%;
  height: 60px;
  margin-right: 10px;
`;

const RoundedImage = styled.Image`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 100px;
`;

const TextInfosContainer = styled.View`
  flex: 4;
`;

const TextInfoTitle = styled.Text`
  font-weight: bold;
  font-size: 17px;
  color: ${WHITE};
  margin-bottom: 3px;
`;
const TextInfoDescription = styled.Text`
  color: ${WHITE};
  font-weight: 200;
`;

interface AvatarProp {
  data: {
    name: string;
    description: string;
  };
}
export default function Avatar({ data }: AvatarProp) {
  return (
    <TopInfosContainer>
      <RoundedImageContainer>
        <RoundedImage source={{ uri: DEFAULT_IMG }}></RoundedImage>
      </RoundedImageContainer>
      <TextInfosContainer>
        <TextInfoTitle>{data.name}</TextInfoTitle>
        <TextInfoDescription>{data.description}</TextInfoDescription>
      </TextInfosContainer>
    </TopInfosContainer>
  );
}

const styles = StyleSheet.create({});
