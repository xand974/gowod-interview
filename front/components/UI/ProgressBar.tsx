import React from "react";
import { StyleSheet, Text, View } from "react-native";
import styled from "styled-components/native";
import { LIGHT_BLUE, DARK } from "assets/styles/@core.style";
import { useState, useEffect } from "react";

const ProgressBarContainer = styled.View`
  margin-bottom: 10px;
  position: relative;
`;

const ProgressBarInactive = styled.View`
  width: 100%;
  height: 5px;
  background-color: ${DARK};
`;

const ProgressBarActive = styled.View`
  width: ${(props: { percentage: number }) => `${props.percentage}%`};
  height: 5px;
  position: absolute;
  background-color: ${LIGHT_BLUE};
`;

interface ProgressBarProp {
  activeValue: number;
  reference: number;
}
export default function ProgressBar({
  activeValue,
  reference,
}: ProgressBarProp) {
  const [percentage, setPercentage] = useState(0);

  useEffect(() => {
    setPercentage(() => {
      return (activeValue / reference) * 100;
    });
  }, [activeValue]);

  return (
    <ProgressBarContainer>
      <ProgressBarInactive></ProgressBarInactive>
      <ProgressBarActive percentage={percentage}></ProgressBarActive>
    </ProgressBarContainer>
  );
}

const styles = StyleSheet.create({});
