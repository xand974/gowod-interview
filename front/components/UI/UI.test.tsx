import renderer from "react-test-renderer";
import Avatar from "./Avatar";
import ProgressBar from "./ProgressBar";

test("render Avatar component", () => {
  const component = renderer
    .create(<Avatar data={{ name: "Alexandre", description: "Malet" }} />)
    .toJSON() as any;

  expect(component.children?.length).toEqual(2);
  expect(component).toMatchSnapshot();
});

test("render Progress Bar component", () => {
  const component = renderer
    .create(<ProgressBar activeValue={10} reference={20} />)
    .toJSON() as any;

  expect(component.children?.length).toEqual(2);
  expect(component).toMatchSnapshot();
});
