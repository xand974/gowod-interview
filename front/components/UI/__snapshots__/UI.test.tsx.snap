// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Avatar component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 60,
          "marginRight": 10,
          "width": "100%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1556817411-31ae72fa3ea0?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 100,
            "height": "100%",
            "objectFit": "cover",
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 4,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginBottom": 3,
          },
        ]
      }
    >
      Alexandre
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontWeight": "200",
          },
        ]
      }
    >
      Malet
    </Text>
  </View>
</View>
`;

exports[`render Progress Bar component 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 10,
        "position": "relative",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#090D16",
          "height": 5,
          "width": "100%",
        },
      ]
    }
  />
  <View
    percentage={0}
    style={
      Array [
        Object {
          "backgroundColor": "#508dd3",
          "height": 5,
          "position": "absolute",
          "width": "0%",
        },
      ]
    }
  />
</View>
`;
